src/main/java/net/sf/jabref/gui/JabRefFrame.java

```diff
@@ -1788,15 +1789,32 @@ public void removeCachedEntryEditors() {
      * This method shows a wait cursor and blocks all input to the JFrame's contents.
      */
     public void block() {
-        getGlassPane().setVisible(true);
+        if (SwingUtilities.isEventDispatchThread()) {
+            getGlassPane().setVisible(true);
+        } else {
+            try {
+                SwingUtilities.invokeAndWait(() -> getGlassPane().setVisible(true));
+            } catch (InvocationTargetException | InterruptedException e) {
+                LOGGER.info("Problem blocking UI", e);
```

Comment:
Makes sense. It is a severe problem if it happens. Still quite unlikely (which was why I selected warn), but clearly no harm in having error. (I'm thinking a lot in risk = severity*probability, but not useful here.)